### ISSUE TITLE
TypeError: locked() got multiple values for argument 'target'

### ISSUE THREAD

**Issue by KarimGeiger** [rytilahti/python-eq3bt#26]
Hi there,

thanks for taking your time to implement and publish this library. Unfortunately, I can't quite get it to work for my system. I'm using [these models](https://smile.amazon.de/gp/product/B014VHFR6Y/), but when trying to connect, I get the following output:

```
# eq3cli --mac 00:1A:22:06:XX:XX --debug
DEBUG:eq3bt.eq3btsmart:Querying the device..
DEBUG:eq3bt.connection:Trying to connect to 00:1A:22:06:XX:XX
DEBUG:eq3bt.connection:Connected to 00:1A:22:06:XX:XX
DEBUG:eq3bt.connection:Writing b'03120a030a151e' to 1041 with with_response=True
DEBUG:eq3bt.connection:Waiting for notifications for 1
DEBUG:eq3bt.connection:Got notification from 1057: b'020108000427'
DEBUG:eq3bt.eq3btsmart:Received notification from the device..
DEBUG:eq3bt.eq3btsmart:Got status: b'020108000427'
DEBUG:eq3bt.eq3btsmart:Parsed status: Container:
    cmd = 2
    mode = Container:
        DST = True
    valve = 0
    target_temp = 19.5
    away = b'' (total 0)
DEBUG:eq3bt.eq3btsmart:Valve state: 0
DEBUG:eq3bt.eq3btsmart:Mode:        auto dst
DEBUG:eq3bt.eq3btsmart:Target temp: 19.5
DEBUG:eq3bt.eq3btsmart:Away end:    None
[00:1A:22:06:12:03] Target 19.5 (mode: auto dst, away: no)
Traceback (most recent call last):
  File "/home/homeassistant/bin/eq3cli", line 11, in <module>
    sys.exit(cli())
  File "/home/homeassistant/lib/python3.6/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/homeassistant/lib/python3.6/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/homeassistant/lib/python3.6/site-packages/click/core.py", line 1114, in invoke
    return Command.invoke(self, ctx)
  File "/home/homeassistant/lib/python3.6/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/homeassistant/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/homeassistant/lib/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/homeassistant/lib/python3.6/site-packages/eq3bt/eq3cli.py", line 38, in cli
    ctx.invoke(state)
  File "/home/homeassistant/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/homeassistant/lib/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/homeassistant/lib/python3.6/site-packages/eq3bt/eq3cli.py", line 161, in state
    ctx.invoke(locked, ctx)
  File "/home/homeassistant/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/homeassistant/lib/python3.6/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/home/homeassistant/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
TypeError: locked() got multiple values for argument 'target'
```

For me, it looks like it is able to connect, since it returns correct information (target_temp, etc.), and I can use some other commands like "temp", but there seems to be an incompatibility somewhere. Sadly, I don't know much about python, so I'm a bit lost here.

Any help is much appreciated!

Thanks

Karim

**Comment by rytilahti** [comment]
That looks weird and I don't know a solution at the moment, but you could comment out the line 161 in `/home/homeassistant/lib/python3.6/site-packages/eq3bt/eq3cli.py` to let it hopefully execute correctly.

**Comment by KarimGeiger** [comment]
Wow, that was a fast response - didn't expect that. Thanks!

Yeah, this seems to solve the problem at hand, but it creates a new one - looks like this product is a little different to the other ones. However, it still works with Home Assistant despite the errors being generated, so for me personally it's not that big of a deal, since that's all I wanted it to do. So: Thanks a lot! I'm finally able to control my thermostats without the app 🎉

For completeness: This is the error I'm getting after commenting out the line:
```
Traceback (most recent call last):
  File "/home/homeassistant/bin/eq3cli", line 11, in <module>
    sys.exit(cli())
  File "/home/homeassistant/lib/python3.6/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/homeassistant/lib/python3.6/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/homeassistant/lib/python3.6/site-packages/click/core.py", line 1114, in invoke
    return Command.invoke(self, ctx)
  File "/home/homeassistant/lib/python3.6/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/homeassistant/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/homeassistant/lib/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/homeassistant/lib/python3.6/site-packages/eq3bt/eq3cli.py", line 38, in cli
    ctx.invoke(state)
  File "/home/homeassistant/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/homeassistant/lib/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/homeassistant/lib/python3.6/site-packages/eq3bt/eq3cli.py", line 162, in state
    ctx.invoke(low_battery, ctx)
  File "/home/homeassistant/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/homeassistant/lib/python3.6/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/home/homeassistant/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
TypeError: low_battery() takes 1 positional argument but 2 were given
```

**Comment by rytilahti** [comment]
Ok, so for some reason click keeps passing those parameters, so this should be fixed somehow indeed, so I'll leave this issue open. Unfortunately I'm not using these thermostats actively and do not have currently time to investigate why that's the case.

Glad that it's working for your use-case though!

**Comment by microraptor** [comment]
I had the same errors and don't think it is the thermostat, but the click library. See PR #28 which fixes the issue.

**Comment by rytilahti** [comment]
Thanks for the PR! I just released 0.1.10 with your fix.

**Comment by TheZoker** [comment]
@rytilathi Can you maybe also create a new release for 1.1.10 on GitHub?
